### PR TITLE
Mi-go

### PR DESCRIPTION
**Pull request by ghost**
:cl: Armhulen
add: Mi-go have entered your realm!
/:cl:

Creatures are now a netherworld inhabitant, but that's just pathing.

 ### MI-GO!
First created by HP Lovecraft and perfected in CDDA, they are crustation alien mimics! obtainable through xenobio at the moment but I may have plans for them later. They mimic noises of all kinds, most of them are just honking, gas, glass, et cet. Some of them might really scare you though, one being a dimension tear. But those are really rare, promise.

### CODE REVIEW

**Comment by tgstation-server** [comment]
You currently have a negative Fix/Feature pull request delta of -2. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

**Comment by Armhulen** [comment]
i made a grammar pr don't delete this

**Comment by RemieRichards** [comment]
I actually sprited Mi-Go before, not the best but better than reusing butterflies lmao.

mi-go

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm
Comment (RemieRichards) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: dont store file paths in strings, they need to be in dm ```'single quotes'``` so if they're removed they throw errors, rather than silently waiting to runtime one day.

**Comment by Armhulen** [comment]
the icon bot is messed up right now here is what my mi-gos look like

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: 'strings/migo_sounds.txt'

like that?

Comment (RemieRichards) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: No.
I mean you can't use a text file.
you have to use a list.

**Comment by praisenarsie** [comment]
close, that grammar pr isn't going to give enough gbp

**Comment by Repukan** [comment]
You currently have a negative Fix/Feature pull request delta of -1. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

**Comment by Armhulen** [comment]
get good

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: do you know how long this list will be mam?

**Comment by duncathan** [comment]
what if I just reach over and...

**Comment by praisenarsie** [comment]
good guy dunc

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm
Comment (RemieRichards) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: don't give a poop, long list is infinitely better than a list that could literally runtime at any moment.

**Comment by Repukan** [comment]
Straight busta

**Comment by Armhulen** [comment]
@praisenarsie fuck you hahaha I WIN HAH AH

@Repukan YOU TOO FUCKER

**Comment by duncathan** [comment]
opened on loan for the other prs

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm
Comment (Xhuis) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: Would `'[soundpath]'` not work for selecting sounds? I recall having done something like that similarly before.

**Comment by Repukan** [comment]
Could you make them spawn exclusively on the hostile reaction.

Makes no sense for hostile nether spawn to spawn docile (or at all but that's something entirely different.)

**Comment by ACCount12** [comment]
I don't see this being any good of a feature. It doesn't add anything to game except for one meh spell and two shitty mobs, and we have more than enough of both already.

**Comment by Armhulen** [comment]
@ACCount12 didn't read the pr

@Repukan not a terrible idea, but xenobiologists are already way too scared to spawn on hostile and the special trait of the mi-go isn't useful for combat, just fucking around

**Comment by Repukan** [comment]
Doesn't fucking matter, if you're adding hostile nether spawn then either go all the way through or just change the PR description to say you're bloating the fucking pool more with non hostile mobs.

They're supposed to be scared to spawn on hostile since, I assume, the strongest mobs can spawn from that pool.

**Comment by Armhulen** [comment]
i think it does (((fucking))) matter, every mob in the mob pool for xenobio is hostile you know that right

**Comment by Armhulen** [comment]
bar xeno maid.

**Comment by Armhulen** [comment]
and also no the best mobs don't spawn from that pool anyways, the nonhostile pool has headslugs and deathbolt carps. also has watchers and fugu, all in nonhostile.

**Comment by imsxz** [comment]
Might wanna reword the description of the PR, people seem to be missing what was actually added and changed here.

I personally think it's a neat change, makes the creatures a lot more interesting than most other melee hostile mobs. The noises were spooky during the play test earlier.

**Comment by Armhulen** [comment]
>adds penguin with a sombrero that's only admin spawn

lol this is a great mob addition to the game!

>adds a netherworld portal with different types of monsters that all do something that the wizard casts and found in xenobio

lol fuckin bloat retard neck yourself

**Comment by ACCount12** [comment]
I wasn't that much against this PR, but OP appears to be retarded. That earns a solid 👎 from me.

**Comment by Armhulen** [comment]
ad hominem is not an argument, you thought repathing a mob was creating an entirely new one, you should just leave

**Comment by Xhuis** [comment]
This is your periodic reminder that scathing comments from people who vanish for years on end or have been continually permabanned with no sign of showing any understanding why mean nothing and can be safely ignored.

**Comment by Armhulen** [comment]
i keep forgetting

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm
Comment (RemieRichards) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: No it would not.
It's a compile time thing.

Feel free to keep ignoring me here @Armhulen it's not my fault when your code breaks.

**Comment by Repukan** [comment]
I'll admit the premise is neat and they're a good design but the main reason I'm opposed to it it's because the game is mostly focused in a demonical kind of setting regarding monsters. You're not only adding two creatures but also the whole netherspawn into the game, which would later seem like a kind of why? Addition if nobody else works or adds on it. Plus it doesn't mesh well with demons since the themes will clash unless you make up some bullshit lore that allows them to spawn on megafauna or in (coming soon Tm) planet station jungle.

I never defended the stupid shamebrero penguin and you're free to look for the PR and check comments to double check that.

Lastly >listening to the bitching of a burnt out echo chamber dweller.

**Comment by Armhulen** [comment]
@Repukan respectable post, i should probably just change it for mi-go to stay in xenobio spawns only because the rest of the monsters aren't as interesting as they are.

**Comment by Repukan** [comment]
Yeah probably, my main gripe is still that this is introducing a new faction (netherworld) without it being fleshed out enough.

**Comment by Armhulen** [comment]
not really, i just wanted to avoid making a new mob file because that would be bloat-y. mi-go has a lot of parenting from creature, and creature was described as an "otherthing" so it seems fair to put it in as an cosmic horror. the actual faction var is mostly for the creatures to not attack the portal they spawn from.

**Comment by Repukan** [comment]
I don't faction in code terms, I mean faction in gameplay knowledge.

Goliaths, hivelords and basilisks were the "asteroid" monsters. Watchers and legions got reskinned to be the "lavaland, demonic" monsters. The hostile drawings belong to the paper wizard, the "drawn" enemies.

I don't know how to explain it in English but I hope this gets the point across.

**Comment by RemieRichards** [comment]
@Repukan Lore faction, is your point.

**Comment by Repukan** [comment]
Ja

**Comment by Armhulen** [comment]
it's not a terrible point though two monsters from a dimension of complete chaos is kinda lame

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm
Comment (Alek2ander) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: its

Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: thank you

**Comment by RemieRichards** [comment]
>Ready

No it's not.

**Comment by Armhulen** [comment]
explain

**Comment by Armhulen** [comment]
oh you're right i didn't remove the wizard spell. silly me!

**Comment by RemieRichards** [comment]
It's a real shame.
Oh, no wait, no it isn't.

**Comment by Armhulen** [comment]
????????????????

**Comment by Xhuis** [comment]
Remie kind of has a tendency to tell people they're doing things wrong and then not elaborate on what things are actually wrong. The only thing I can think of is the sound file choosing, which seems remedied now, and both testing and test*merges* have shown it works fine.

**Comment by RemieRichards** [comment]
I have elaborated, twice, and both of you (including you @Xhuis) even responded to it????
You also just stated what it is, so you clearly know what I'm talking about.

An No, it is not resolved.

**Comment by Xhuis** [comment]
I was moreso agitated by not elaborating to Arm, who is still a new an inexperienced coder. The changes you outlined were in the `Outdated diff` dropdown, which to the majority of people signals that something is no longer an issue.

Last time I tried to make a long list of strings so as not to store them in different files, I was shouted down by maintainers, meaning the precedent already set goes against the norm of putting them in single quotes. There doesn't seem to be an easy way to win here.

**Comment by RemieRichards** [comment]
list of strings != list of filepaths.
```''``` is designed for filepaths, and at compile time warns you if those files are missing.

**Comment by Xhuis** [comment]
Would it be possible at all to use external files to store the mi-go sounds in a more efficient way than a huge static list of filepaths?

Also dang, that was a fast response.

**Comment by RemieRichards** [comment]
Not without losing the compile time protection.
What you're suggesting is just dynamically building this list, which is what's happening at the moment, and is wrong to do so.

**Comment by Xhuis** [comment]
Thanks, byond!

**Comment by RemieRichards** [comment]
Well, yes actually, thanks byond, not a lot of languages have compile time protection for 'resource' files such as sprites and sounds, where byond does.

it's just stupid to ignore it where useful, being ignorant of the feature is fine but being reminded twice is just a whole different kettle of fish.

**Comment by Xhuis** [comment]
I wouldn't want to willfully ignore it - I just thought that building a list from a file would be much more efficient, especially given the fact that the errors are both fixed in the same way (by adding a file in that location where none exists.) I do have to wonder if it's possible to do something in that way, building a list of asset locations based on an external file.

**Comment by RemieRichards** [comment]
You're missing the point, there's no softcode solution that will tell you at compile time a file you're referencing is missing, the best you can do is ```fexist()s``` checks in list-making procs, which doesn't fix the issue and just like map related stuff no one will be bothered to actually fix the errors said ```fexist()``` checks would shout out.

**Comment by Xhuis** [comment]
I guess the end question would be if the added cleanliness of having the files separately is worth the possibility of causing runtime errors in lieu of compile warnings, which I would say is subjective. Basing your code on the assumption that nobody will bother to fix the errors is silly and would arbitrarily limit what could be accomplished with it, in my eyes.

**Comment by RemieRichards** [comment]
I disagree and we're not going to come to an agreement so, this is over.

**Comment by Xhuis** [comment]
Fine by me.

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: TRUE

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: TRUE

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: i believe this is false, doesn't gold_core_spawnable = 2 mean it spawns on hostile reaction only?

**Comment by Armhulen** [comment]
image
he really showed me

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm
Comment (ShizCalev) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: Strike that then, will clean that up in another PR.

**Comment by duncathan** [comment]
Remie is right please listen to her

**Comment by Armhulen** [comment]
i'll do it but don't come complaining

**Comment by Armhulen** [comment]
son of a bitch who just conflicted me

**Comment by ShizCalev** [comment]
it-was-me-jim-carrey-liar-liar

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm
Comment (ShizCalev) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: HOSTILE_SPAWN

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: No_spawn

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: tgstation.dme
Comment (ShizCalev) [inline tgstation.dme]: Uwut

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: tgstation.dme
Comment (Armhulen) [reply tgstation.dme]: fuck ummmm

Comment (Armhulen) [reply tgstation.dme]: i think i accidentally included it in the merge conflict fixing my bad

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: code\modules\mob\living\simple_animal\hostile\netherworld.dm:59:error: No_spawn: undefined var

Comment (ShizCalev) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: on a phone, all caps.

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm
Comment (ACCount12) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: Is this the largest string in our codebase?

Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: i didn't want this either

File: code/modules/spells/spell_types/conjure.dm
Comment (ShizCalev) [inline code/modules/spells/spell_types/conjure.dm]: FALSE

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm
Comment (RemieRichards) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: Oh my fucking god did you even LISTEN to what you were told to do?
1. ```/static``` list, there need only be a single copy of this per all mi-go.
2. SINGLE QUOTES, ```'``` not DOUBLE ```"``` or else the entire thing is pointless.

Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: ummmmmmmmm

File: code/modules/spells/spell_types/conjure.dm
Comment (DoctoreMad) [inline code/modules/spells/spell_types/conjure.dm]: Please try harder

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm
Comment (Xhuis) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: Hey, at least he tried, y'know?

**Comment by zamolxius** [comment]
make them spawn enclosed in asteroids

File: code/modules/spells/spell_types/conjure.dm
Comment (Armhulen) [reply code/modules/spells/spell_types/conjure.dm]: whatever, it's an admin spell that's completely unbalanced.

Comment (Armhulen) [reply code/modules/spells/spell_types/conjure.dm]: i mean look above that spawns like 6 securitrons with magic missiles if i remember correctly

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: that's it i've had it you're getting a confused emoji under your post

**Comment by Armhulen** [comment]
@RemieRichards WHY DOESN'T IT WORK I DID WHAT YOU ASKED

**Comment by RemieRichards** [comment]
Because, SPOILERS, it's doing what I said it would.
It's compile time warning you about bad file paths.

WHO WOULD HAVE THUNK IT.

**Comment by Armhulen** [comment]
what's so bad about those file paths

**Comment by RemieRichards** [comment]
There's no file! That's a pretty fucking bad file path if you ask anyone.

**Comment by Armhulen** [comment]
like, from typos? ok, i'll take another look i suppose

**Comment by Xhuis** [comment]
Hey, don't knock those file paths. They put in their best effort.

**Comment by ShizCalev** [comment]
@Armhulen doesn't compile.

**Comment by Armhulen** [comment]
i'll be in coderbus later today to fix this and get it out the door

**Comment by epochayur** [comment]
excited to see this merged

**Comment by Armhulen** [comment]
ok DONE

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm
Comment (ShizCalev) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: new(loc) since it's a transformation.

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm
Comment (AnturK) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: this check before prob

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm
Comment (AnturK) [inline code/modules/mob/living/simple_animal/hostile/netherworld.dm]: Standard declaration please.

File: strings/migo_sounds.txt
Comment (AnturK) [inline strings/migo_sounds.txt]: Remove this file if it's unused.

**Review by AnturK — CHANGES_REQUESTED** [review]

File: strings/migo_sounds.txt
Comment (Armhulen) [reply strings/migo_sounds.txt]: oh right

File: code/modules/mob/living/simple_animal/hostile/netherworld.dm
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/netherworld.dm]: ?

**Comment by Armhulen** [comment]
@AnturK no idea what standard declaration is here, do you mean Say() instead of say()?
anyways the rest is all done and ready

**Comment by Xhuis** [comment]
I think he means `say(message)`, but I'm not sure.

**Comment by Substance455** [comment]
Armhulen borrowed the sprite from one of CDDA's community made tilesets. Please credit the original spriter.

image

**Comment by AnturK** [comment]
What the hell armhulen, is the sprite set license compatible ? If not this whole thing is going to get reverted.

**Comment by Substance455** [comment]
You can use the sprite. Just don't say its yours and credit the OG artist or I'll get really mad and make the revert happen.

**Comment by Armhulen** [comment]
ok, will do. sorry for that mess. just edit the changelog in a new pr, yeah?

**Comment by kevingranade** [comment]
You need crediting, and a notice that the asset is licensed under CC-BY-SA.
See: https://github.com/CleverRaven/Cataclysm-DDA/blob/master/LICENSE.txt

**Comment by ninjanomnom** [comment]
@kevingranade crediting got fixed quickly after this pr was merged and all our assets are marked CC-BY-SA by default.

**Comment by Iatots** [comment]
Was this found out as a consequence of deitus getting fucking destroyed by mi-gos in cata DDA, and proceeding to have a PTSD attack when one was spawned on station?

**Comment by Bawhoppen** [comment]
We need better sprite review requirements overall. We didn't even have WJ actually review the sprite for quality, despite having the sprite tag on it.

**Pull request merged**
